S3-BOX: add procedure on customizing the status images (#30302)
home-assistant/home-assistant.github.io: source/voice_control/s3-box-customize.markdown

1. Create your own images according the specifications defined [above](#about-the-images).
2. Copy all 6 images into a folder. For example: `voice_assistant_pictures`.
3. Make sure you have [access to your configuration files](/common-tasks/os/#configuring-access-to-files).
   - [Install the Samba add-on](/common-tasks/os/#installing-and-using-the-samba-add-on).
   - This allows you to copy paste multiple files at once.
4. Copy your image folder into the configuration folder:
   - Open the `config` folder and open the `ESPHome` folder.
   - Copy your image folder in there.
   ![ESP32-S3-BOX-3 config file](/images/assist/s32-s3-add-image-folder.png)

### To add your images to the configuration

1. In Home Assistant, go to [**Settings** > **Add-ons** > **ESPHome**](https://my.home-assistant.io/redirect/supervisor_addon/?addon=5c53de3b_esphome), and **Open Web UI**.
2. On the **ESP32-S3-BOX-3** add-on, select edit.
   - **Result**: An editor opens, showing the configuration file.
   ![ESP32-S3-BOX-3 config file](/images/assist/esp32-adopt-s3-01.png)

3. To add your images, add the following lines into the `substitution` block.

   ```yaml
   substitution:

>      loading_illustration_file: /config/esphome/voice_assistant_pictures/loading_320_240.png

Reviewer: Make all these paths relative to the folder that contains the ESPHome configuration. 
```suggestion
     loading_illustration_file: ./voice_assistant_pictures/loading_320_240.png
```

>      idle_illustration_file: /config/esphome/voice_assistant_pictures/idle_320_240.png

Reviewer: Clean up the file names so it's more readable, we don't need width/height in there. 

We should also standardize the filenames according to what we use on GitHub, here in our examples and in our example picture.

Before you can import new illustrations, you need to install the ESPHome add-on and adopt the device in the add-on.

1. Make sure the ESP32-S3-BOX-3 is up and running and connected to your Wi-Fi.
2. Go to **Settings** > **Add-ons** and check if you have the **ESPHome** add-on installed.
   - If you haven't done so, go to [**Settings** > **Add-ons** > **ESPHome**](https://my.home-assistant.io/redirect/supervisor_addon/?addon=5c53de3b_esphome) to install the add-on.
3. Start the add-on and select **Open web UI**.
4. In the ESPHome add-on dashboard, on the **ESP32-S3-BOX-3** card, select **Adopt**.

   ![Adopt the ESP32-S3-BOX-3 in the ESPHome add-on](/images/assist/esp32-adopt-s3.png)
5. If you like, give it a new name. Then, select **Adopt**.
   - Adopting an ESPHome device allows us to customize the existing software.
   - **Result**: The status will change to **Online**.
6. Now that you have set up the ESPHome add-on, continue with one of the 2 different methods to add custom images:
   - [Using images from a community repository](#using-images-from-a-community-repository)
   - [Using your own images](#using-your-own-images)

## Option 1: Using images from a community repository

> If you want new images but don't want to create your own, you can use images by the community.

Reviewer 2: ```suggestion
If you want new images but don't want to create your own, you can use images from the community.
```

       idle_illustration_file: https://github.com/jlpouffier/home-assistant-s3-box-community-illustrations/raw/main/frenck/illustrations/idle.png
       listening_illustration_file: https://github.com/jlpouffier/home-assistant-s3-box-community-illustrations/raw/main/frenck/illustrations/listening.png
       thinking_illustration_file: https://github.com/jlpouffier/home-assistant-s3-box-community-illustrations/raw/main/frenck/illustrations/thinking.png
       replying_illustration_file: https://github.com/jlpouffier/home-assistant-s3-box-community-illustrations/raw/main/frenck/illustrations/replying.png
       error_illustration_file: https://github.com/jlpouffier/home-assistant-s3-box-community-illustrations/raw/main/frenck/illustrations/error.png
     ```

4. Save the changes and select **Install**:
   - Depending on your environment, the installation process can take a while.
![ESP32-S3-BOX-3 config file](/images/assist/esp32-s3-config-05.png)
   
5. Once the installation is complete, you can see the new image on the ESP32-S3-BOX-3.

>    - Now, speak a command to test the new setting. For example, *OK Nabu turn off the living room lights*.

Reviewer 2: ```suggestion
   - Now, speak a command to test the new setting. For example, *OK Nabu, turn off the living room lights*.
```

## Option 2: Using your own illustrations

There are 2 parts to this:

- [Preparing your own illustrations](#to-prepare-your-own-images)
- [Adding your illustrations to the configuration](#to-add-your-images-to-the-configuration)

### About the image specifications

Here's what you need to know to get the best result on your ESP32-S3-BOX-3 screen.

#### Using light and dark image background

In the [Chart](#esp32-s3-box-3-voice-assistant-status-illustrations), you can see that the default images use different background colors. This is to make it easier to recognize a state change when you look at your screen.

In your images, you could use 2 different background colors:

- For loading and idle: use a dark background

> - For the listening, thinking, replying: use a light background

Reviewer 2: ```suggestion
- For listening, thinking, and replying: use a light background
```

- For error: As you like
  
If your images have transparency, you can define the background color in the configuration. The procedure below shows how to change the background.

#### Image dimensions and file format

- **Dimensions**: The screen is 320 x 240 pixels. If the image you provide is not in a 4:3 ratio, the background color will be shown in the remaining area.

> - **File format**: png, jpeg, or svg

Reviewer 2: ```suggestion
- **File format**: PNG, JPEG, or SVG
```

     error_illustration_file: voice_assistant_gfx/error.png
   ```

4. If you used transparency in your images and you want to change the background color, add the following lines into the `substitution` block:
   - The `000000` stands for black, `FFFFFF` stands for white in [hexadecimal color code](https://www.w3schools.com/html/html_colors_hex.asp).
   - If you want to use different colors, replace the corresponding color code.
   - To find the color code, you can use a tool like the Google color picker.

     ```yaml
     substitution:
       ...
       loading_illustration_background_color: '000000'
       idle_illustration_background_color: '000000'
       listening_illustration_background_color: 'FFFFFF'
       thinking_illustration_background_color: 'FFFFFF'
       replying_illustration_background_color: 'FFFFFF'
       error_illustration_background_color: '000000'
     ```

5. Save the changes and select **Install**.
   ![ESP32-S3-BOX-3 config file](/images/assist/s32-s3-add-image-config-02.png)
6. Save the changes.
   - Depending on your environment, the installation process can take a while.
7. Once the installation is complete, you can see the new image on the S3-BOX.

>    - Now, speak a command to test the new setting. For example, *OK Nabu turn on the light*.

Reviewer 2: ```suggestion
   - Now, speak a command to test the new setting. For example, *OK Nabu, turn on the light*.
```

---
title: "Customize the S3-BOX with your own images"
product_name: ESP32-S3-BOX
device_name_entry: ESP32-S3-BOX
config_link: /voice_control/thirteen-usd-voice-remote/#to-delete-the-atom-echo-configuration-from-esphome
---

> This tutorial will show you how to replace the Home Assistant status images on the ESP32-S3-BOX by custom images. 

Reviewer: Include a video of the Frenck box here.

Reviewer 3: @balloob Can you past the youtube link once uploaded? We don't have the rights

      thinking_illustration_file: https://raw.githubusercontent.com/jlpouffier/home-assistant-s3-box-community-illustrations/main/frenck/illustrations/thinking_320_240.png
      replying_illustration_file: https://raw.githubusercontent.com/jlpouffier/home-assistant-s3-box-community-illustrations/main/frenck/illustrations/replying_320_240.png
      error_illustration_file: https://raw.githubusercontent.com/jlpouffier/home-assistant-s3-box-community-illustrations/main/frenck/illustrations/error_320_240.png
    ```

   {% endraw %}

4. Save the changes and select **Install**:
   - Depending on your environment, the installation process can take a while.
   - Expect this to take about 25 minutes. The code needs to be compiled before it can be installed.
![ESP32-S3-BOX config file](/images/assist/esp32-s3-config-05.png)
   
5. Once the installation is complete, you can see the new image on the S3-BOX.
   - Reboot the device.
   - Now, speak a command to test the new setting. For example, *OK Nabu turn on the light*.

## Using your own images

Perform this task if you want to use your own images on the S32-S3-BOX.

> This task consist of 2 procedures:

Reviewer: Can we use the word `steps` instead, as it sounds less formal.

- [Preparing your own images](#to-prepare-your-own-images)

> - [Adding your images to the configuration](#to-add-your-images-to-the-configuration)

Reviewer: Why are these links? Do we want users to jump to the next step? Wouldn' they lose context.

Author: Yes, we want them to go to either one or the other procedure.

> **Duration**: ca. 35 minutes, without counting the time to create the images.

Reviewer: ```suggestion
**Duration**: ca. 35 minutes, excluding the time to create the images.
```

### About the images

> To make sure the images can be displayed properly on the S3-BOX screen, check out the following requirements.

Reviewer: We are talking about pictures so we should show example pictures to make it clear what we are talking about. Create an overview picture that includes the six pictures of our default assistant.

Reviewer 3: Use this picture if you want:
![image](https://github.com/home-assistant/home-assistant.io/assets/5878296/da89c27e-1b34-4de3-9b85-0e39f543e146)

#### Illustrating 6 different states

There are 6 different states to illustrate:

- loading
- idle
- listening
- thinking
- replying
- error

#### Using light and dark image background

To make it easier to distinguish the different states when you look at your screen, change the background depending on the state:

- For loading and idle: use a dark background
- For the listening, thinking, replying: use a light background
- For error: As you like
  

> You can also save the images with transparent background and then change the background in the configuration file.

Reviewer: This adds more complexity. Maybe we can move this to some advanced section at the bottom ?

Author: That is true. 
On the other hand, I would like to have everything there is to know about the images in this section. So that people realize already at this point that they could save the image with a transparent background.
Rephrased it a bit to make it easier to read. 
Will come back to this again later and restructure if needed.

#### Image dimensions and file format

- **Screen dimensions**: 320 x 240 pixels
- **File format**: png

### To prepare your own images

1. Create your own images according the specifications defined [above](#about-the-images).

> 2. Paste all 6 images into a folder. For example: `voice_assistant_pictures`.

Reviewer: Paste is a bit weird. Use word "copy"

3. Make sure you have [access to your configuration files](/common-tasks/os/#configuring-access-to-files).
   - [Install the Samba add-on](/common-tasks/os/#installing-and-using-the-samba-add-on).
   - Samba allows you to copy paste multiple files at once.
4. Paste your image folder into the configuration folder:

>    - Using Samba, open the `config` folder and open the `ESPHome` folder.

Reviewer: You don't "Use samba" on any of the major OS. You use network shares.

   - Paste your image folder in there.
   ![ESP32-S3-BOX config file](/images/assist/s32-s3-add-image-folder.png)

### To add your images to the configuration

1. Go to **Settings** > **Add-ons**, select the **ESPHome** add-on and **Open Web UI**.
2. On the **ESP32-S3-BOX** add-on, select edit.
   - **Result**: An editor opens, showing the configuration file.
   ![ESP32-S3-BOX config file](/images/assist/esp32-adopt-s3-01.png)

3. To add your images, add the following lines into the `substitution` block.
    {% raw %}

     ```yaml

>       loading_illustration_file: /config/esphome/voice_assistant_pictures/loading_320_240.png

Reviewer: indentation lost here too.

---
title: "Customize the S3-BOX with your own images"
product_name: ESP32-S3-BOX
device_name_entry: ESP32-S3-BOX
config_link: /voice_control/thirteen-usd-voice-remote/#to-delete-the-atom-echo-configuration-from-esphome
---

This tutorial will show you how to replace the Home Assistant status images on the ESP32-S3-BOX by custom images. 

You can either prepare your own images or use images from a community repository.
It applies to the following three ESP32 devices by Espressif:

- [ESP32-S3-BOX](https://www.espressif.com/en/news/ESP32-S3-BOX_video)
- [ESP32-S3-BOX-3](https://www.espressif.com/en/news/ESP32-S3-BOX-3)
- [ESP32-S3-BOX-Lite](https://www.espressif.com/en/news/ESP32-S3-BOX_video)

> The steps are the same for all three product variants. The term *ESP32-S3-BOX* is used to refer to all three variants.

Reviewer: You can only buy the ESP32 S3 Box 3 today. Users can't buy other ones. I feel like we make it more complicated than it should be by showing all those other variants this prominently.

## Prerequisites

- Latest version of Home Assistant, installed with the Home Assistant Operating System
- [Home Assistant Cloud](https://www.nabucasa.com) or a manually configured [Assist Pipeline](/voice_control/voice_remote_local_assistant)
- Successfully completed the [ESP32-S3-BOX voice assistant](/voice_control/s3_box_voice_assistant/) tutorial

## Adopting the device in the ESPHome add-on

Before you can import new images, you need to install the ESPHome add-on and adopt the device in the add-on.

1. Make sure the ESP32-S3-BOX is up and running and connected to your Wi-Fi.
2. Go to **Settings** > **Add-ons** and check if you have the **ESPHome** add-on installed.

>    - If you haven't done so, install the add-on.

Reviewer: Add a My link to link/install the add-on

Reviewer 3: This is the link FYI : https://my.home-assistant.io/redirect/supervisor_addon/?addon=5c53de3b_esphome

And this is the markdown to integrate the MY button

[![Open your Home Assistant instance and show the dashboard of an add-on.](https://my.home-assistant.io/badges/supervisor_addon.svg)](https://my.home-assistant.io/redirect/supervisor_addon/?addon=5c53de3b_esphome)

Reviewer 3: ```
[![Open your Home Assistant instance and show the dashboard of an add-on.](https://my.home-assistant.io/badges/supervisor_addon.svg)](https://my.home-assistant.io/redirect/supervisor_addon/?addon=5c53de3b_esphome)
```

3. Start the add-on.
4. Once the add-on is running, select **Open web UI**.
5. In the ESPHome add-on dashboard, on the **ESP32-S3-BOX** card, select **Adopt**.
   ![Adopt the ESP32-S3-BOX in the ESPHome add-on](/images/assist/esp32-adopt-s3.png)
6. If you like, give it a new name. Then, select **Adopt**.

> 7. Copy the encryption key and store it in a safe place.

Reviewer: Probably no need for this and makes it more complicated? 

Home Assistant will pick up encryption keys automatically because you're using the add-on. Encryption keys are stored in the board config, which is set in the YAML files and is included in core backups.

   - It will also be available within the UI, but it is always good to keep a copy.
   - **Result**: Once you copied the key, the status will change to **Online**.

> 8. Now that you have set up the ESPHome add-on, continue with one of the 2 different methods to add custom images:

Reviewer: Users have set up the add-on + adopted the configuration. Should we explain what adoption is?

> Adopting an ESPHome device allows us to customize the existing software on the device.

   - [Using images from a community repository](#using-images-from-a-community-repository)
   - [Using your own images](#using-your-own-images)

## Using images from a community repository

- Perform this procedure, if you want to use ready-made images by the community.

> - If you want to use your own images, skip this procedure and go to [Using your own images](#using-your-own-images) instead.

Reviewer: I would drop the bullets here. It's the introduction to this section.

**Duration**: about 30 minutes

### To use images from the community

1. On the **ESP32-S3-BOX** add-on, select edit.
   - **Result**: An editor opens, showing the configuration file.
   ![ESP32-S3-BOX config file](/images/assist/esp32-adopt-s3-01.png)
2. For inspiration, we have prepared some images for you.

>    - Check them out in this [repository](https://github.com/jlpouffier/home-assistant-s3-box-community-illustrations/tree/main).

Reviewer: "Check them out" is ambiguous. Maybe use word 'Browse'

> 3. For this tutorial, we will use some images of Frenck.

Reviewer: should we link to the Frenck folder so user can see from the readme where the below values comes from ?

   - Add the following lines into the `substitution` block.

   {% raw %}

     ```yaml

>       loading_illustration_file: https://raw.githubusercontent.com/jlpouffier/home-assistant-s3-box-community-illustrations/main/frenck/illustrations/loading_320_240.png

Reviewer: The way this is embedded misses the indentation that is required to put it under the substitutions block, resulting in bad YAML.

I also am not sure if users know what "substitution" block is. We should maybe include `substitution:` in the example YAML here ?

Reviewer 3: If you can try again with the URL I provided on the repo, that would be nice. 
They work for me, I double checked again

```yaml
substitutions:
  loading_illustration_file: https://github.com/jlpouffier/home-assistant-s3-box-community-illustrations/raw/main/frenck/illustrations/loading_320_240.png
  idle_illustration_file: https://github.com/jlpouffier/home-assistant-s3-box-community-illustrations/raw/main/frenck/illustrations/idle_320_240.png
  listening_illustration_file: https://github.com/jlpouffier/home-assistant-s3-box-community-illustrations/raw/main/frenck/illustrations/listening_320_240.png
  thinking_illustration_file: https://github.com/jlpouffier/home-assistant-s3-box-community-illustrations/raw/main/frenck/illustrations/thinking_320_240.png
  replying_illustration_file: https://github.com/jlpouffier/home-assistant-s3-box-community-illustrations/raw/main/frenck/illustrations/replying_320_240.png
  error_illustration_file: https://github.com/jlpouffier/home-assistant-s3-box-community-illustrations/raw/main/frenck/illustrations/error_320_240.png
  ```

      idle_illustration_file: https://raw.githubusercontent.com/jlpouffier/home-assistant-s3-box-community-illustrations/main/frenck/illustrations/idle_320_240.png
      listening_illustration_file: https://raw.githubusercontent.com/jlpouffier/home-assistant-s3-box-community-illustrations/main/frenck/illustrations/listening_320_240.png
      thinking_illustration_file: https://raw.githubusercontent.com/jlpouffier/home-assistant-s3-box-community-illustrations/main/frenck/illustrations/thinking_320_240.png
      replying_illustration_file: https://raw.githubusercontent.com/jlpouffier/home-assistant-s3-box-community-illustrations/main/frenck/illustrations/replying_320_240.png
      error_illustration_file: https://raw.githubusercontent.com/jlpouffier/home-assistant-s3-box-community-illustrations/main/frenck/illustrations/error_320_240.png
    ```

>    {% endraw %}

Reviewer: ![CleanShot 2023-12-12 at 12 43 01](https://github.com/home-assistant/home-assistant.io/assets/1444314/69ae8067-a843-41da-bf4d-1e9595f60b1f)
I don't know where these 2 artifacts come from on the page, maybe because of the extra new lines here? Why are we using `raw` tags ?

   
4. Save the changes and select **Install**:
   - Depending on your environment, the installation process can take a while.

>    - Expect this to take about 25 minutes. The code needs to be compiled before it can be installed.

Reviewer: This is not going to take 25 minutes?

>    - Expect this to take about 25 minutes. The code needs to be compiled before it can be installed.

Reviewer: These 2 lines say the same thing, let's consolidate ?

![ESP32-S3-BOX config file](/images/assist/esp32-s3-config-05.png)
   
5. Once the installation is complete, you can see the new image on the S3-BOX.

>    - Reboot the device.

Reviewer: The device is automatically rebooted.

>    - Now, speak a command to test the new setting. For example, *OK Nabu turn on the light*.

Reviewer: This one only works if the user has put the device in an area

Reviewer 3: You can use "Turn off the living room lights" I think it's the most common command as it's the most common area

## Using your own images

> Perform this task if you want to use your own images on the S32-S3-BOX.

Reviewer: Can we be a bit more enthusiasm and energy in this sentence ? Also, it's ESP32-S3-BOX (missing ES)

This task consist of 2 procedures:

- [Preparing your own images](#to-prepare-your-own-images)
- [Adding your images to the configuration](#to-add-your-images-to-the-configuration)
  
**Duration**: ca. 35 minutes, without counting the time to create the images.

### About the images

To make sure the images can be displayed properly on the S3-BOX screen, check out the following requirements.

#### Illustrating 6 different states

There are 6 different states to illustrate:

- loading
- idle
- listening
- thinking
- replying
- error

#### Using light and dark image background

To make it easier to distinguish the different states when you look at your screen, change the background depending on the state:

- For loading and idle: use a dark background
- For the listening, thinking, replying: use a light background
- For error: As you like
  
You can also save the images with transparent background and then change the background in the configuration file.

#### Image dimensions and file format

> - **Screen dimensions**: 320 x 240 pixels

Reviewer: This is not a requirement. ESPHome compilation will automatically make it fit in this space. If it is not 4:3 ratio, it will show the background where there is no picture drawn.

> - **File format**: png

Reviewer: I don't think that this is a requirement.

Reviewer: Confirmed. Not a requirement

### To prepare your own images

1. Create your own images according the specifications defined [above](#about-the-images).
2. Paste all 6 images into a folder. For example: `frenck`.
3. Make sure you have [access to your configuration files](/common-tasks/os/#configuring-access-to-files).
   - [Install the Samba add-on](/common-tasks/os/#installing-and-using-the-samba-add-on).
   - Samba allows you to copy paste multiple files at once.
4. Paste your image folder into the configuration folder:
   - Using Samba, open the `config` folder and open the `ESPHome` folder.
   - Paste your image folder in there.
   ![ESP32-S3-BOX config file](/images/assist/s32-s3-add-image-folder.png)

### To add your images to the configuration

1. Go to **Settings** > **Add-ons**, select the **ESPHome** add-on and **Open Web UI**.

> 2. On the **ESP32-S3-BOX** add-on, select edit.

Reviewer: It's not an add-on

   - **Result**: An editor opens, showing the configuration file.
   ![ESP32-S3-BOX config file](/images/assist/esp32-adopt-s3-01.png)

3. For this tutorial, we will use some images of Frenck.
   - To add your images, add the following lines into the `substitution` block.
    {% raw %}

     ```yaml
      loading_illustration_file: /config/esphome/frenck/loading_320_240.png
      idle_illustration_file: /config/esphome/frenck/idle_320_240.png
      listening_illustration_file: /config/esphome/frenck/listening_320_240.png
      thinking_illustration_file: /config/esphome/frenck/thinking_320_240.png
      replying_illustration_file: /config/esphome/frenck/replying_320_240.png
      error_illustration_file: /config/esphome/frenck/error_320_240.png
      ```

    {% endraw %}

4. If you used transparency in your images and you want to change the background color, add the following lines:

>    - The default is black and white.

Reviewer: A background is a single color. Do you mean that defaults differ per state?

Reviewer 3: The default background colors are

- Black for the loading state
- Black for the idle state
- White for the listening state
- White for the thinking state
- White for the replying state
- Black for the error state

Author: I removed any mention of defaults for now. I'm not sure it is even needed.

>    - If you want to use different colors, replace the corresponding hexadecimal color code.

Reviewer: Should we link to a website where a user can understand what hexadecimal color codes are and click on colors to get them ?

Should we explain 000000 is black and ffffff is white ?

Reviewer 3: The best is to ask them to look for "Color Picker" on Google.
They have a nice widget directly on the search page

<img width="692" alt="image" src="https://github.com/home-assistant/home-assistant.io/assets/5878296/d9a2705d-5fe1-484c-8e12-7c2294604e71">

     {% raw %}

     ```yaml
      loading_illustration_background_color: '000000'
      idle_illustration_background_color: '000000'
      listening_illustration_background_color: 'FFFFFF'
      thinking_illustration_background_color: 'FFFFFF'
      replying_illustration_background_color: 'FFFFFF'
      error_illustration_background_color: '000000'
      ```

   {% endraw %}
5. Save the changes and select **Install**.
   ![ESP32-S3-BOX config file](/images/assist/s32-s3-add-image-config-02.png)

> 6. Save the changes and restart Home Assistant.

Reviewer: You don't need to restart Home Assistant

   - Depending on your environment, the installation process can take a while.

>    - Expect this to take about 25 minutes. The code needs to be compiled before it can be installed.

Reviewer: Should we explain the word "compilation" or can we just talk about creating your customized voice assistant ?

Reviewer: Also, subsequent installs are faster.

7. Once the installation is complete, you can see the new image on the S3-BOX.

>    - Reboot the device.

Reviewer: Device is rebooted as part of installation
